gotopkg.go

```diff
@@ -1,204 +0,0 @@
-package main
-
-import (
-	"flag"
-	"fmt"
-	"github.com/groob/plist"
-	"io"
-	"io/ioutil"
-	"log"
-	"os"
-	"os/exec"
-	"strings"
-	"syscall"
-	"time"
-)
-
-const RECIPE_FAILED_CODE = 70
-
-type AutopkgRunReport struct {
-	Failures []struct {
-		Recipe    string `plist:"recipe"`
-		Message   string `plist:"message"`
-		Traceback string `plist:"traceback"`
-	} `plist:"failures"`
-	SummaryResults struct {
-		URLDownloaderSummaryResult struct {
-			DataRows []struct {
-				DownloadPath string `plist:"download_path"`
-			} `plist:"data_rows"`
-			Header      []string `plist:"header"`
-			SummaryText string   `plist:"summary_text"`
-		} `plist:"url_downloader_summary_result"`
-		MunkiImporterSummaryResult struct {
-			DataRows []struct {
-				PkgRepoPath string `plist:"pkg_repo_path"`
-				Catalogs    string `plist:"catalogs"`
-				Version     string `plist:"version"`
-				PkginfoPath string `plist:"pkginfo_path"`
-				Name        string `plist:"name"`
-			} `plist:"data_rows"`
-			Header      []string `plist:"header"`
-			SummaryText string   `plist:"summary_text"`
-		} `plist:"munki_importer_summary_result"`
-	} `plist:"summary_results"`
-}
-
-func (rpt *AutopkgRunReport) DecodePlist(r io.Reader) error {
-	err := plist.NewDecoder(r).Decode(&rpt)
-	if err != nil {
-		return fmt.Errorf("error decoding report from reader:", err)
-	}
-
-	return nil
-}
-
-var autopkg string
-
-func main() {
-	flagAutopkg := flag.String("autopkg", "autopkg", "path to autopkg")
-	flag.Parse()
-
-	path, err := exec.LookPath(*flagAutopkg)
-	if err != nil {
-		log.Fatalf("error finding autopkg binary: %s", err)
-	}
-
-	autopkg = path
-	log.Printf("using autopkg: %s\n", autopkg)
-
-	if flag.NArg() < 1 {
-		log.Fatal("must specify recipe arguments")
-	}
-
-	continuousRun(flag.Args())
-
-	os.Exit(0)
-}
-
-const TIME_DELAY time.Duration = time.Second * 2
-const RUN_EVERY time.Duration = time.Second * 15
-
-type recipeRunStatus struct {
-	name        string
-	lastStarted time.Time
-	firstRun    bool
-}
-
-func continuousRun(recipeNames []string) {
-	var recipes []recipeRunStatus
-
-	for _, recipeName := range recipeNames {
-		recipes = append(recipes, recipeRunStatus{recipeName, time.Now(), false})
-	}
-
-	var lastSince time.Duration
-
-	for {
-		for i, recipe := range recipes {
-			lastSince = time.Since(recipe.lastStarted)
-			if !recipe.firstRun || lastSince > RUN_EVERY {
-				recipes[i].firstRun = true
-				recipes[i].lastStarted = time.Now()
-				log.Printf("running recipe %s (first run or last run > %s [%s])", recipe.name, RUN_EVERY, lastSince)
-
-				autopkgRunOneRecipeCheckDLFirst(recipe.name, false)
-			}
-		}
-
-		log.Printf("sleeping %s", TIME_DELAY)
-		time.Sleep(TIME_DELAY)
-	}
-}
-
-func autopkgRunOneRecipeCheckDLFirst(recipe string, forceFull bool) {
-	var report AutopkgRunReport
-
-	// run, but only check first
-	if !forceFull {
-		output, err := autopkgRun([]string{"-v", "-c", recipe}, &report)
```

Comment:
what if we redefined `autopkgrRun` as

`func autopkgRun(params []string) (output []byte, report *AutopkgRunReport, err error)`

or even 

`func autopkgRun(params ...string) (output []byte, report *AutopkgRunReport, err error)`

The variadic(...) input means "zero or more" so now `autopkrRun()` and `autopkrRun("-v")` are both valid.